src/androidMain/kotlin/software/momento/kotlin/sdk/internal/InternalDataClient.android.kt

```diff
@@ -2,97 +2,158 @@ package software.momento.kotlin.sdk.internal
 
 import com.google.protobuf.ByteString
 import grpc.cache_client.ECacheResult
+import grpc.cache_client._DeleteRequest
 import grpc.cache_client._GetRequest
-import grpc.cache_client._GetResponse
 import grpc.cache_client._SetRequest
-import io.grpc.Metadata
-import software.momento.kotlin.sdk.utils.ValidationUtils
+import software.momento.kotlin.sdk.internal.utils.ValidationUtils
 import software.momento.kotlin.sdk.auth.CredentialProvider
 import software.momento.kotlin.sdk.config.Configuration
 import software.momento.kotlin.sdk.exceptions.CacheServiceExceptionMapper
 import software.momento.kotlin.sdk.exceptions.InternalServerException
+import software.momento.kotlin.sdk.internal.utils.ByteStringExtensions.toByteString
+import software.momento.kotlin.sdk.responses.cache.DeleteResponse
 import software.momento.kotlin.sdk.responses.cache.GetResponse
 import software.momento.kotlin.sdk.responses.cache.SetResponse
-import java.io.Closeable
 import kotlin.time.Duration
 
 internal actual class InternalDataClient actual constructor(
-    credentialProvider: CredentialProvider, configuration: Configuration
-): Closeable {
-
+    credentialProvider: CredentialProvider, configuration: Configuration, private val itemDefaultTtl: Duration
+) : InternalClient() {
     private val stubsManager: DataGrpcStubsManager
 
     init {
-        stubsManager = DataGrpcStubsManager(credentialProvider)
+        stubsManager = DataGrpcStubsManager(credentialProvider, configuration.transportStrategy.grpcConfiguration)
     }
 
     internal actual suspend fun set(
-        cacheName: String, key: String, value: String, ttl: Duration
+        cacheName: String, key: String, value: String, ttl: Duration?
+    ): SetResponse = validateAndSendSet(cacheName, key.toByteString(), value.toByteString(), ttl)
+
+    internal actual suspend fun set(
+        cacheName: String, key: ByteArray, value: String, ttl: Duration?
+    ): SetResponse = validateAndSendSet(cacheName, key.toByteString(), value.toByteString(), ttl)
+
+    internal actual suspend fun set(
+        cacheName: String, key: String, value: ByteArray, ttl: Duration?
+    ): SetResponse = validateAndSendSet(cacheName, key.toByteString(), value.toByteString(), ttl)
+
+    internal actual suspend fun set(
+        cacheName: String, key: ByteArray, value: ByteArray, ttl: Duration?
+    ): SetResponse = validateAndSendSet(cacheName, key.toByteString(), value.toByteString(), ttl)
```

Comment:
is it possible to do the conversions to ByteArray at the CacheClient level? might help reduce the amount of code duplication between CacheClient and InternalDataClient

Reply:
We're converting to a ByteString, which is implementation specific. We could add an extra conversion to ByteArray first to simplify it, but that would be an extra copy we don't need to do otherwise. I like the idea though.